lib/bolt/target.rb

```diff
@@ -86,6 +86,10 @@ def set_resource(resource)
     end
     # rubocop:enable Naming/AccessorMethodName
 
+    def resource(type, title)
```

Comment:
I don't _think_ this version is needed, since we don't expose `resource()` as a function on the data type.

Reply:
Having it here enables `$target.resource(type, title)` - we do this for set_resources, I think it makes sense to for `resource` as well

Reply:
The difference is that the `set_resources` function _calls_ `Target#set_resource` directly. Nothing is calling this method. `$target.resource(type, title)` works because that's just how function invocation works in Puppet. It will call a function named `resource`, passing in `$target` as the first argument.

_If_ there were a `resource` function defined on the Target datatype, then it would call that 
instead, which would dispatch to this method. But since that's defined on the datatype, this method should never be invoked.